File: Test_ChickenEgg.cpp
#include "ChickenEgg.cpp"
#include <gtest/gtest.h>
#include <string>

using namespace std;

TEST(ChickenEggTest, AllFunctionalityofChickenEgg){
    ChickenEgg testCE(30, "Rembo");
    ASSERT_EQ("Rembo", testCE.getProducer());
    ASSERT_EQ(CHICKEN_EGG, testCE.getObjectType());
    ASSERT_EQ(FARMPRODUCT, testCE.getProductType());
    ASSERT_EQ(30, testCE.getPrice());
}

File: Java/src/test/java/engisubs/gameobject/product/farmproduct/egg/TestChickenEgg.java
package engisubs.gameobject.product.farmproduct.egg;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;
public class TestChickenEgg{
    private ChickenEgg test;
    @Before
    public void initObject(){
        test = new ChickenEgg(25, "Name");
    }

    @Test
    public void instantiationTest(){
        Assert.assertEquals("Name", test.getProducer());
        Assert.assertEquals(GameObjectType.CHICKEN_EGG, test.getObjectType());
        Assert.assertEquals(ProductType.FARMPRODUCT, test.getProductType());
        Assert.assertEquals(25, test.getPrice());
    }

}